presto-pinot/src/main/java/com/facebook/presto/pinot/PinotClusterInfoFetcher.java

```diff
@@ -0,0 +1,201 @@
+    private static final String ROUTING_TABLE_API_TEMPLATE = "http://%s/debug/routingTable/%s";
+    private static final String TIME_BOUNDARY_API_TEMPLATE = "http://%s/debug/timeBoundary/%s";
+
+    private static final Logger log = Logger.get(PinotClusterInfoFetcher.class);
+
+    private static final CloseableHttpClient HTTP_CLIENT = HttpClients.createDefault();
+    private final String controllerUrl;
+    private final String zookeeperServers;
+    private DynamicBrokerSelector dynamicBrokerSelector;
+    private String instanceId = "Presto_pinot_master";
+    private ObjectMapper objectMapper = new ObjectMapper();
+
+    @Inject
+    public PinotClusterInfoFetcher(PinotConfig pinotConfig)
+            throws SocketException, UnknownHostException
+    {
+        this(pinotConfig.getZookeeperUrl(), pinotConfig.getPinotCluster(), pinotConfig.getControllerUrl());
+    }
+
+    public PinotClusterInfoFetcher(String zookeeperUrl, String pinotCluster, String controllerUrl)
+            throws SocketException, UnknownHostException
+    {
+        log.info("Trying to init PinotClusterInfoFetcher with Zookeeper: %s, PinotCluster %s, ControllerUrl: %s.", zookeeperUrl, pinotCluster, controllerUrl);
+        zookeeperServers = zookeeperUrl + "/" + pinotCluster;
+        instanceId = instanceId + "_" + NetUtil.getHostAddress();
+        this.controllerUrl = controllerUrl;
+    }
+
+    public static String sendHttpGet(final String url)
+            throws Exception
+    {
+        HttpUriRequest request = RequestBuilder.get(url).setHeader(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON).build();
+        return HTTP_CLIENT.execute(request, getStringResponseHandler());
+    }
+
+    private static ResponseHandler<String> getStringResponseHandler()
+    {
+        return new ResponseHandler<String>()
+        {
+            @Override
+            public String handleResponse(HttpResponse response)
+                    throws ClientProtocolException, IOException
+            {
+                int status = response.getStatusLine().getStatusCode();
+                HttpEntity entity = response.getEntity();
+                if (familyForStatusCode(response.getStatusLine().getStatusCode()) == HttpStatus.Family.SUCCESSFUL && entity != null) {
+                    return EntityUtils.toString(entity);
+                }
+                else {
+                    if (entity == null) {
+                        throw new ClientProtocolException("Entity is null");
+                    }
+                    throw new ClientProtocolException("Unexpected status: " + status);
+                }
+            }
+        };
+    }
+
+    public void close()
+            throws IOException
+    {
+        HTTP_CLIENT.close();
+    }
+
+    private String getControllerUrl()
+    {
+        return this.controllerUrl;
+    }
+
+    @SuppressWarnings("unchecked")
+    public List<String> getAllTables()
+            throws Exception
```

Comment:
no exception is thrown

Reply:
Exception is thrown here. I guess you meant `getBrokerHost`. Removed it there.